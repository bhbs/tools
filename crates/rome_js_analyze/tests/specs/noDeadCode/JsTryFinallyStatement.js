function JsTryFinallyStatement1() {
    try {
        test();
    } catch (err) {
        test();
    } finally {
        test();
        return;
    }

    afterFinallyReturn();
}
