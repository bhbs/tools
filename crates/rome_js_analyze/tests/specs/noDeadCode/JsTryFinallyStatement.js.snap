---
source: crates/rome_js_analyze/tests/spec_tests.rs
expression: JsTryFinallyStatement.js
---
# Input
```js
function JsTryFinallyStatement1() {
    try {
        test();
    } catch (err) {
        test();
    } finally {
        test();
        return;
    }

    afterFinallyReturn();
}

```

# Diagnostics
```
warning[noDeadCode]: This code is unreachable
   ┌─ JsTryFinallyStatement.js:11:5
   │
 8 │         return;
   │         ------- This statement will return from the function ...
   ·
11 │     afterFinallyReturn();
   │     --------------------- ... before it can reach this code


```


